src/Microsoft.ML.AutoML/API/AutoCatalog.cs

```diff
@@ -538,55 +541,158 @@ public AutoMLExperiment CreateExperiment()
+
+            var option = new OneHotOption
+            {
+                InputColumnNames = inputColumnNames,
+                OutputColumnNames = outputColumnNames,
+            };
+
+            return new SweepableEstimator[] { SweepableEstimatorFactory.CreateOneHotEncoding(option), SweepableEstimatorFactory.CreateOneHotHashEncoding(option) };
         }
 
         /// <summary>
         /// Create a single featurize pipeline according to <paramref name="data"/>. This function will collect all columns in <paramref name="data"/> and not in <paramref name="excludeColumns"/>,
-        /// featurizing them using <see cref="CatalogFeaturizer(string, string)"/>, <see cref="NumericFeaturizer(string, string)"/> or <see cref="TextFeaturizer(string, string)"/>. And combine
+        /// featurizing them using <see cref="CatalogFeaturizer(string[], string[])"/>, <see cref="NumericFeaturizer(string[], string[])"/> or <see cref="TextFeaturizer(string, string)"/>. And combine
         /// them into a single feature column as output.
         /// </summary>
         /// <param name="data">input data.</param>
         /// <param name="catalogColumns">columns that should be treated as catalog. If not specified, it will automatically infer if a column is catalog or not.</param>
+        /// <param name="numericColumns">columns that should be treated as numeric. If not specified, it will automatically infer if a column is catalog or not.</param>
+        /// <param name="textColumns">columns that should be treated as text. If not specified, it will automatically infer if a column is catalog or not.</param>
         /// <param name="outputColumnName">output feature column.</param>
         /// <param name="excludeColumns">columns that won't be included when featurizing, like label</param>
-        internal MultiModelPipeline Featurizer(IDataView data, string outputColumnName = "Features", string[] catalogColumns = null, string[] excludeColumns = null)
+        public MultiModelPipeline Featurizer(IDataView data, string outputColumnName = "Features", string[] catalogColumns = null, string[] numericColumns = null, string[] textColumns = null, string[] excludeColumns = null)
         {
-            throw new NotImplementedException();
+            // validate if there's overlapping among catalogColumns, numericColumns, textColumns and excludeColumns
+            var overallColumns = new string[][] { catalogColumns, numericColumns, textColumns, excludeColumns }
+                                    .Where(c => c != null)
+                                    .SelectMany(c => c);
+
+            if (overallColumns != null)
+            {
+                Contracts.Assert(overallColumns.Count() == overallColumns.Distinct().Count(), "detect overlapping among catalogColumns, numericColumns, textColumns and excludedColumns");
```

Comment:
Nit: changed `detect overlapping` to `detected overlapping`.

I'm also personally a fan of the oxford comma.

Reply:
## Resolved